addons/preview-project-description/userscript.js

```diff
@@ -0,0 +1,171 @@
+export default async function ({ addon, console, msg }) {
+  const enableSwitcher = document.createElement("button");
+  const enableSwitcherText = document.createElement("span");
+  enableSwitcher.id = "sa-preview-notes-instructions";
+  enableSwitcher.classList.add("button", "action-button", "sa-preview-desc");
+  enableSwitcher.addEventListener("click", () => togglePreview());
+  enableSwitcher.appendChild(enableSwitcherText);
+
+  let currentlyEnabled = false;
+  let wasEverEnabled = false;
+  let currentlyRerendering = false;
+
+  let avoidInfiniteLoops = 0; // Just in case, make sure we aren't introducing an infinite loop (unsure if Redux considers this)
+  addon.tab.redux.initialize();
+  addon.tab.redux.addEventListener("statechanged", (e) => {
+    // We don't want a state change to disable the preview mode.
+    if (!addon.tab.redux.state.scratchGui.mode.isPlayerOnly) return;
+    if (e.detail.action._saProjectDescription) return;
+    if (avoidInfiniteLoops > 5) console.log("Avoiding an infinite loop");
+    else if (currentlyEnabled) {
+      const num = ++avoidInfiniteLoops;
+      // Restore preview mode.
+      queueMicrotask(() => enablePreview());
+      setTimeout(() => {
+        if (avoidInfiniteLoops === num) avoidInfiniteLoops = 0;
+      }, 0);
+    }
+  });
+
+  while (true) {
+    // Same waitForElement call as animated-thumb/userscript.js
+    await addon.tab.waitForElement(".flex-row.subactions > .flex-row.action-buttons", {
+      markAsSeen: true,
+      reduxCondition: (state) => state.scratchGui.mode.isPlayerOnly,
+    });
+
+    // TODO: also change animated-thumb/userscript.js
```

Comment:
Reminder: TODO

Reply:
Maybe I should create a `waitForSelfProject()` utility that makes sure to consider going to your own project logged out and logging-in within the same session (Scratch does not reload in that case)